content/en/docs/tasks/tools/install-kubectl.md

```diff
@@ -295,63 +295,111 @@ kubectl cluster-info dump
+The below sections walk you through installing bash-completion and setting up kubectl autocompletion for both Linux and macOS.
 
-```shell
-echo "source <(kubectl completion bash)" >> ~/.bashrc
-```
+#### Bash on Linux
 
-### On macOS, using bash
-On macOS, you will need to install bash-completion support via [Homebrew](https://brew.sh/) first:
+bash-completion is available in many package managers (see [here](https://github.com/scop/bash-completion#installation)). You can install it, for example, with `apt-get install bash-completion` or `yum install bash-completion`.
 
-```shell
-## If running Bash 3.2 included with macOS
-brew install bash-completion
-## or, if running Bash 4.1+
+The above commands create `/usr/share/bash-completion/bash_completion`, which is the main script of bash-completion. Depending on your package manager, you have to manually source this file in your `~/.bashrc` file.
+
+To find out, restart your shell and run `type _init_completion`. If the command succeeds, you're already set, otherwise add the following to your `~/.bashrc` file:
+
+~~~bash
+source /usr/share/bash-completion/bash_completion
+~~~
+
+Restart your shell and verify that bash-completion is correctly installed with `type _init_completion`.
+
+You now need to take care of sourcing the kubectl completion script in your shell sessions. There are two ways in which you can do this:
+
+- Source the completion script in your `~/.bashrc` file:
+
+    ~~~bash
+    echo 'source <(kubectl completion bash)' >>~/.bashrc
+    ~~~
+
+- Add the completion script to the `/etc/bash_completion.d` directory:
+
+    ~~~bash
+    kubectl completion bash >/etc/bash_completion.d/kubectl
+    ~~~
+
+*Note that bash-completion sources all completion scripts in `/etc/bash_completion.d`.*
+
+Both approaches are equivalent. After restarting your shell, autocompletion should be working.
+
+#### Bash on macOS
+
+The default version of Bash on macOS is 3.2, and unfortunately, the kubectl completion script only works with Bash 4.1+. One way around this is to install a newer version of Bash on macOS (see instructions [here](https://itnext.io/upgrading-bash-on-macos-7138bd1066ba)).
+
+If you upgraded your version of Bash, you can install bash-completion with Homebrew:
+
+~~~bash
 brew install bash-completion@2
-```
+~~~
 
-Follow the "caveats" section of brew's output to add the appropriate bash completion path to your local .bashrc.
+*Note that the `@2` stands for bash-completion 2, which is required for the kubectl completion script. In turn, bash-completion 2 requires Bash 4.1+, that's why you needed to upgrade Bash.*
 
-If you installed kubectl using the [Homebrew instructions](#install-with-homebrew-on-macos) then kubectl completion should start working immediately.
+Follow the "Caveats" section of the `brew install` command. In particular add the following lines to your `~/.bashrc` or `~/.bash_profile` file:
```

Comment:
where is this "Caveats" section?

also i think you want to bring back `[Homebrew instructions](#install-with-homebrew-on-macos)`

Reply:
I changed the formulation. The "Caveats" section refers to the output of `brew install` where instructions to add the two lines below to `~/.bashrc` or `~/.bash_profile` are given.

I also added the link.